kolibri/core/auth/models.py

```diff
@@ -75,11 +75,28 @@
 from kolibri.core.errors import KolibriValidationError
 from kolibri.core.fields import DateTimeTzField
 from kolibri.core.fields import JSONField
+from kolibri.core.utils.validators import JSON_Schema_Validator
 from kolibri.plugins.app.utils import interface
 from kolibri.utils.time_utils import local_now
 
 logger = logging.getLogger(__name__)
 
+extra_fields_schema = {
+    "type": "object",
+    "additionalProperties": False,
+    "properties": {
+        "facility": {"type": "object"},
+        "on_my_own_setup": {"type": "boolean"},
+        "pin_code": {"type": "string"},
```

Comment:
If the `pin_code` is supposed to be a 4-digit number/integer, why is the type `string`?

Reply:
The discussion here is what prompted my decision to use a string instead https://github.com/learningequality/kolibri/pull/9993#discussion_r1082758692. But I am also open to here any particular wins we could benefit from using `int` that may not be available with string.

Reply:
Thanks for sharing that context, that reasoning makes sense to me for the input field. 

Thinking about the API separately though, it seems it likely accepts a JSON post body? If so, one benefit of the JSON body is that you can disambiguate between strings and numbers, since it's JSON, unlike with normal form-encoded post bodies. Of course, your API validation enforces that the string is essentially a number so aligning the parameter type and schema could make the API a bit clearer without needing the entire context of how the pin code is input or validated.

Reply:
Thanks for this insight. Makes sense!

Reply:
I have updated the code to take into consideration these recommendations. Thanks again @bjester